server/agent/conversation/manager.py

```diff
@@ -43,6 +43,11 @@ def respond_to_user_message(self, conversation_id: int, data_set_id: int, user_i
                                               role='user',
                                               text=message)
 
+        # Set the conversation name if it's the first message
+        if not conversation.name:
+            conversation.name = user_message.text[:255]
```

Comment:
Maybe just rewrite the whole first message here? I don't think that would be a big problem